From f87ee53ca82e270e6d3cbea8e5672b068e450d79 Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Sun, 26 Apr 2015 10:54:34 +0300
Subject: [PATCH] Redirect wpa_supplicant output in wpa_start() (closes #112)

wpa_supplicant>=2.4 sets stdout to be line-buffered. This breaks
wpa_supplicant_scan() since the startup message from wpa_supplicant
("Successfully initialized wpa_supplicant") gets printed before the
essids filename.
---
 src/lib/wpa | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/wpa b/src/lib/wpa
index e8ce8b1..b5fda8a 100644
--- a/src/lib/wpa
+++ b/src/lib/wpa
@@ -80,7 +80,7 @@ wpa_start() {
     fi
 
     do_debug wpa_supplicant -B -P "$pidfile" -i "$interface" -D "$driver" \
-                            "$configuration" $WPAOptions
+                            "$configuration" $WPAOptions > /dev/null
 
     # Wait up to one second for the pid file to appear
     if ! timeout_wait 1 '[[ -f $pidfile ]]'; then
-- 
2.3.6

